### ISSUE TITLE
Error when trying to deploy the backend-frontend sample project

### ISSUE THREAD

**Issue by mtammacco** [dotnet/tye#942]
I got this error when I tried to run "tye deploy --interactive" with the backend-frontend sample provided:
YamlDotNet.Core.YamlException: (Line: 6, Col: 5, Idx: 119) - (Line: 6, Col: 6, Idx: 120): Expected 'MappingStart', got 'SequenceStart' 

Step To Reproduce:
- Go to the backend-frontend sample project
- Run tye deploy --interactive
- During the "Validating secret" phase I got this exception:

An unhandled exception has occurred, how unseemly:
YamlDotNet.Core.YamlException: (Line: 6, Col: 5, Idx: 119) - (Line: 6, Col: 6, Idx: 120): Expected 'MappingStart', got 'SequenceStart' (at Line: 6, Col: 5, Idx: 119).
   at YamlDotNet.Core.ParserExtensions.Require[T](IParser parser)
   at YamlDotNet.Serialization.NodeDeserializers.DictionaryNodeDeserializer.DeserializeHelper(Type tKey, Type tValue, IParser parser, Func`3 nestedObjectDeserializer, IDictionary result)
   at YamlDotNet.Serialization.NodeDeserializers.DictionaryNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
   at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
   at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
   at YamlDotNet.Serialization.NodeDeserializers.CollectionNodeDeserializer.DeserializeHelper(Type tItem, IParser parser, Func`3 nestedObjectDeserializer, IList result, Boolean canUpdate)
   at YamlDotNet.Serialization.NodeDeserializers.CollectionNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
   at YamlDotNet.Serialization.NodeDeserializers.EnumerableNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass3_0.<DeserializeValue>b__0(IParser r, Type t)
   at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.Deserializer.Deserialize(IParser parser, Type type)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](IParser parser)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](TextReader input)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](String input)
   at k8s.Yaml.LoadFromString[T](String content)
   at k8s.Yaml.LoadFromStreamAsync[T](Stream stream)
   at k8s.KubernetesClientConfiguration.LoadKubeConfigAsync(FileInfo kubeconfig, Boolean useRelativePaths)
   at k8s.KubernetesClientConfiguration.BuildConfigFromConfigFileAsync(FileInfo kubeconfig, String currentContext, String masterUrl, Boolean useRelativePaths)
   at k8s.KubernetesClientConfiguration.BuildConfigFromConfigFile(FileInfo kubeconfig, String currentContext, String masterUrl, Boolean useRelativePaths)
   at k8s.KubernetesClientConfiguration.BuildConfigFromConfigFile(String kubeconfigPath, String currentContext, String masterUrl, Boolean useRelativePaths)
   at k8s.KubernetesClientConfiguration.BuildDefaultConfig()
   at Microsoft.Tye.ValidateSecretStep.ExecuteAsync(OutputContext output, ApplicationBuilder application, ServiceBuilder service) in /_/src/Microsoft.Tye.Core/ValidateSecretStep.cs:line 57
   at Microsoft.Tye.ApplicationExecutor.ExecuteAsync(ApplicationBuilder application) in /_/src/Microsoft.Tye.Core/ApplicationExecutor.cs:line 37
   at Microsoft.Tye.Program.ExecuteDeployAsync(OutputContext output, ApplicationBuilder application, String environment, Boolean interactive, Boolean force) in /_/src/tye/Program.DeployCommand.cs:line 107
   at Microsoft.Tye.Program.<>c.<<CreateDeployCommand>b__7_0>d.MoveNext() in /_/src/tye/Program.DeployCommand.cs:line 57
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Invocation.CommandHandler.GetResultCodeAsync(Object value, InvocationContext context)
   at System.CommandLine.Invocation.ModelBindingCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseParseErrorReporting>b__19_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseMiddleware>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass14_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseDebugDirective>b__10_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseExceptionHandler>b__0>d.MoveNext()

- tye --version
   0.6.0-alpha.21070.5+a42e4463943e3136dbd1de38474d8d62b802797c
- The platform: macOS

Any ideas?
Thanks.

**Comment by jkotalik** [comment]
Can you share your tye.yaml file?

**Comment by jkotalik** [comment]
Looks like we could somehow be generating invalid yaml as part of the ValidateSecretStep? Besides that. not sure.

**Comment by mtammacco** [comment]
Thanks a lot.
Here is my tye.yaml file:

name: microservices
services:
- name: backend
  project: backend/backend.csproj
- name: frontend
  project: frontend/frontend.csproj
- name: redis
  image: redis
  bindings:
    - port: 6379
      connectionString: "${host}:${port}"
- name: redis-cli
  image: redis
  args: "redis-cli -h redis MONITOR"

**Comment by mtammacco** [comment]
Hi,
if I remove the Redis dependency in tye.yaml it works. The problem seems to be in the validation secrets.

**Comment by skiboutz** [comment]
Hi @mtammacco ,
I have the same problem and I figured it was because of one the dependencies of Tye ([KubernetesClient](https://www.nuget.org/packages/KubernetesClient/)). Tye uses version 2.0.17 and I was able to reproduce the same problem in a console app until I upgraded to 4.0.8 (latest is 4.0.10 as of right now)

I'm not sure but it might also be related to the version of kubernetes I am using (I'm using K8s 1.20 with minikube 1.17).

@jkotalik could you look into upgrading the KubernetesClient in Tye or confirm requires a specific Kubernetes version at the moment?

**Comment by mtammacco** [comment]
Hi @skiboutz,
thank you very much for your response.
How could I update the [KubernetesClient](https://www.nuget.org/packages/KubernetesClient/) in Tye?

These are the versions used:
minikube v1.17.1 on Darwin 10.15.7
Kubernetes v1.20.2 on Docker 20.10.2
tye version: 0.6.0-alpha.21070

**Comment by skiboutz** [comment]
Hi @mtammacco ,
Locally I had to clone Tye's repo, update the dependencies and then building the tool locally. I then uninstall tye with the following command `dotnet tool uninstall -g Microsoft.Tye`. 
I then installed my Tye version with the following command `dotnet tool install --global --add-source **<path to local tool>** --version <tool version>`
where **<path to local tool>** is the path to the directory containing your local tool and <tool version> is the version in the generated package (0.7.0-dev in my case).

So until the Tye team fixes the dependency in their repo it's the only way I was able to complete the tutorial. I might do I pull request to fix it but I'm a little busy right now so it might take a while.

I hope this helps
